source/common/common/logger.cc

```diff
@@ -14,14 +14,17 @@ namespace Logger {
 
 #define GENERATE_LOGGER(X) Logger(#X),
 
-std::vector<Logger> Registry::all_loggers_ = {ALL_LOGGER_IDS(GENERATE_LOGGER)};
-
 Logger::Logger(const std::string& name) {
   logger_ = std::make_shared<spdlog::logger>(name, Registry::getSink());
   logger_->set_pattern("[%Y-%m-%d %T.%e][%t][%l][%n] %v");
   logger_->set_level(spdlog::level::trace);
 }
 
+std::vector<Logger>& Registry::allLoggers() {
+  static std::vector<Logger> all_loggers = {ALL_LOGGER_IDS(GENERATE_LOGGER)};
```

Comment:
This should be `* all_loggers = new std::vector<logger>(initializers)`, otherwise you face the static destruction order fiasco.